client/service.go

```diff
@@ -464,6 +512,17 @@ func (s *Service) dispatcher(client *Client, fail chan struct{}) bool {
 	}
 }
 
+func (s *Service) subs() []packet.Subscription {
```

Comment:
Can you inline this function? It's unnecessary to create another slice. You can just use `v.(packet.Subscription)' above.